src/provider/states/create_config.rs

```diff
@@ -0,0 +1,296 @@
+                    .to_string(),
+            };
+            fail_fatal!(e);
+        }
+        let container = containers[0].clone();
+
+        if let Some(volumes) = _pod.volumes() {
+            debug!("Found {} volumes in pod {}", volumes.len(), _pod.name());
+            if let Some(mounts) = container.volume_mounts() {
+                debug!("Found {} mounts in pod {}", mounts.len(), _pod.name());
+                // Got mounts and volumes, we can now decide which ones we need to act upon
+                for mount in mounts {
+                    for volume in volumes {
+                        if mount.name.eq(&volume.name) {
+                            let target_dir = target_directory.join(&mount.mount_path);
+                            if let Some(config_map) = &volume.config_map {
+                                if let Some(map_name) = &config_map.name {
+                                    if let Ok(map) = self
+                                        .retrieve_config_map(client.clone(), map_name.to_string())
+                                        .await
+                                    {
+                                        debug!("found config map: {:?} - applying", config_map);
+                                        match self.apply_config_map(
+                                            map,
+                                            target_dir,
+                                            &CreatingConfig::create_render_data(pod_state),
+                                        ) {
+                                            Err(e) => {
+                                                return Transition::next(
+                                                    self,
+                                                    SetupFailed {
+                                                        message:
+                                                            "Failed to create file from config map!"
+                                                                .to_string(),
+                                                    },
+                                                );
+                                            }
+                                            _ => {}
+                                        }
+                                    }
+                                }
+                            } else {
+                                warn!("Skipping volume {} - it is not a config map", volume.name);
+                            }
+                        }
+                    }
+                }
+            };
+        }
+        debug!("Transitioning to service creation");
+        Transition::next(self, CreatingService)
+    }
+
+    async fn json_status(
+        &self,
+        _pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> anyhow::Result<serde_json::Value> {
+        make_status(Phase::Pending, &"status:initializing")
```

Comment:
fix string and/or give no reason at all

And the "&" before the string shouldn't be needed?